### PR TITLE
KOGITO-4978 : Make new nodes Editable

### PR DESCRIPTION
**Pull request by inodeman**
Hi @romartin please find this PR for this ticket. 

It also fixes KOGITO-5036 about issue with Imported nodes being editable

Artifact can be found here. https://drive.google.com/drive/folders/1zF8JxV0phIgRFJktqFwiUp3SjITc0oEy?usp=sharing

### CODE REVIEW

**Comment by inodeman** [comment]
Jenkins do fdb

**Comment by caponetto** [comment]
Hi @inodeman,
We are using GitHub workflows for our CI, so Jenkins commands won't work here.
Please consider rebasing your branch with the `main` branch in order to get the latest update on the workflows.
Also note that `CI :: Full Downstream Build` outputs a vsix file based on your changes.

**Comment by inodeman** [comment]
Hi @caponetto thanks for the tip

**Comment by domhanak** [comment]
@jomarko Hello there, could you please take a look at this one and see if this does not break some tests of yours for DMN? I remember you had tests that were adding nodes and this might potentially interfere.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java
Comment (domhanak) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java:42]: Please split this into two test cases -> testIsEditableImported, testIsEditableNotImported as they are clearly different units

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java
Comment (domhanak) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java:148]: curious, when can this happen? When can `DomGlobal.window` be null?

**Comment by domhanak** [comment]
I found one issue, not sure it is related. Let me know.
*Summary*
Names of nodes are blank in DMN editors when switching to different tab e.g. DRD actions.

1. Create a dmn with couple of nodes
2. Add new DRD actions to the node
3. Navigate back to the decision canvas
4. The nodes names are blank

Screenshot from 2021-05-18 11-29-09

**Comment by romartin** [comment]
Good catch @domhanak !! :+1: 
We are lookign at this thing, looks a bug we have to fix.

**Comment by jomarko** [comment]
I didn't found other issue than mentioned by Dominik https://github.com/kiegroup/kogito-editors-java/pull/26#issuecomment-843013145.

Removing myself from reviewers list.

**Comment by romartin** [comment]
@inodeman sent a proposal for the DMN fix, see https://gist.github.com/romartin/ad4b3c0272ac850c233de9986b125e33

Also please rebase and try to get builds ready for testing again, once possible

Thanks!!

**Comment by inodeman** [comment]
Hi @romartin applied the new approach and rebased. Can you review it? Thanks a lot

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java
Comment (domhanak) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java:42]: still valid

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java
Comment (domhanak) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java:66]: Again you are testing multiple units it one test. If for example `assertEquals(true, inline.isFiltered(userTask));` fails others asserts will not be run.

Comment (inodeman) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java:66]: Hi @domhanak split now into two tests.

**Comment by inodeman** [comment]
Hi @domhanak @romartin ready again for review. Thanks

**Comment by inodeman** [comment]
VSIX extension can be found here. https://drive.google.com/drive/folders/1zF8JxV0phIgRFJktqFwiUp3SjITc0oEy?usp=sharing

**Comment by domhanak** [comment]
@inodeman Hi I was able to run into this issue:
Screenshot from 2021-06-02 08-20-45

Just keep adding nodes using quick menu and sometimes the previous node gets blanked. Could you please take a look?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java
Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java:66]: I do not see this being split.

Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-client/src/test/java/org/kie/workbench/common/stunner/bpmn/client/canvas/controls/BPMNInlineTextEditorControlTest.java:66]: Please test each node separately

**Comment by inodeman** [comment]
Hi @domhanak let me look at both issues, thanks for reporting

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java
Comment (inodeman) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/canvas/controls/inlineeditor/DMNCanvasInlineTextEditorControlTest.java:42]: Hi @domhanak now split.

**Comment by inodeman** [comment]
Hi @domhanak I fixed issue about Text being on top of Explore/Properties Panel, the other issue, out of 200 times I created a node I was able to reproduce 1 for some reason, I made a change, and have not been able to reproduce it since, can you please tell me if it is still visible to you?

**Comment by domhanak** [comment]
@inodeman Thanks, will take a look.

**Comment by domhanak** [comment]
So I am still able to reproduce the issue number 1 - when I use quick menu the source node name does get empty, but in very rare cases so I am not sure what is wrong here. I can report it for future investigations.

The second issue is now fixed, the name is not overlapping properties and it is hidden under the panel, but a second issue emerges from this.


1. Have a node next to properties panel and use quick menu to add a new node (BPMN, DMN) 
2. The new node is added and completely hidden under properties (canvas was expanded)
3. Use scrollbar to get the new node visible.
4. The new node has no default name and it is empty.

See the video:
https://user-images.githubusercontent.com/5937251/120976693-e1988380-c772-11eb-9a69-49425800889c.mp4

**Comment by inodeman** [comment]
Hi @domhanak @romartin I have fixed side effect with fix on issue #2, I will focus on issue #1 to make sure it is not reproducible anymore

**Comment by inodeman** [comment]
Hi @domhanak @romartin tested latest changes, and as far as I can tell, the #1 issue has been greatly reduced. Can you review it?

**Comment by romartin** [comment]
Hey @domhanak , please missing reivew iteration again here. Thanks! (me too)

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java:157]: Hi @inodeman , it think it's also good tyring to drop that null check fo the Window object, if possible. You should run the tests by using the `@LienzoMockitoTestRunner`, this way I'm pretty sure the window will not be null. Are you running tests cases using this runner? Also otherwise check other tests what they're doing with the window object, but I think when using the lienzo runner the instance is being set by default.

**Review by romartin — COMMENTED** [review]
Code looks good to me @inodeman , just commented one thing as I agree iwht @domhanak on the null window check. 

@inodeman @domhanak thanks for this work and testing, let's see if we can move this PR forward please as start closing things :)

**Comment by domhanak** [comment]
*Acceptance criteria*

*Edit mode must be actived once creating a new node, even from:
** palette :heavy_check_mark: 
** toolbox :heavy_check_mark: 
*All new nodes, except following onces, will behave this way: :heavy_check_mark: 
** Connectors - it's not possible to use name editor popup acutally for connectors, so this new feature cannot yet activate the edition for connectors
** Gateway, Start Events, End Events, Intermediate Events
** DMN elements from the externals models
** Applicable for both BPMN and DMN editors :heavy_check_mark: 

The issues reported are no reproducible anymore LGTM. Please address @romartin comments about null window check.
Also my comment about merging multiple unit tests - https://github.com/kiegroup/kogito-editors-java/pull/26/files#r640595675 - is still not addressed. So good to go when the comments are addressed.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java
Comment (domhanak) [reply kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/inlineeditor/InlineTextEditorBoxViewImpl.java:157]: +1 checks like this just for testing purposes are not a way to go.

**Comment by inodeman** [comment]
Hi @domhanak tests are now slipt, @romartin I removed the null check, tried with LienzoMockitoTestRunner but still window was nill, so I instead changed the tests so that the tests "doNothing" when calling that method, can you guys review it?

**Review by romartin — APPROVED** [review]
Thanks guys for your collaboration here. I think now it's everything ok. Thanks @inodeman for also fixing that small details in comments :)

So approving! Once @domhanak approves, plz let me know and I'll merge.

Thanks!

**Review by domhanak — APPROVED** [review]
Thanks

**Pull request merged**
